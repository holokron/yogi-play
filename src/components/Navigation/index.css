.navigation {
    font-size: 13px !important;
}