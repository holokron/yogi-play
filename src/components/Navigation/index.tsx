import * as React from 'react'
import Navbar from 'reactstrap/lib/Navbar'
import NavbarToggler from 'reactstrap/lib/NavbarToggler'
import Collapse from 'reactstrap/lib/Collapse'
import './index.css'
import AuthMenu from '../AuthMenu'
import MainMenu from '../MainMenu'
import ToggleContainer from '../../containers/ToggleContainer'
import NavBrandLink from '../../components/NavBrandLink'
import TextToSpeech from '../TextToSpeech'

export interface Props {
}

export interface State {
    toggled: boolean
}

export default class Navigation extends React.PureComponent<Props, State> {
    state: State = {
        toggled: false
    }

    constructor(props: Props) {
        super(props)

        this.closeMenu = this.closeMenu.bind(this)
        this.toggleMenu = this.toggleMenu.bind(this)
    }

    closeMenu() {
        this.setState(() => ({
            toggled: false,
        }))
    }

    toggleMenu() {
        this.setState(({ toggled }: State) => ({
            toggled: !toggled,
        }))
    }

    render() {
        return (
            <Navbar dark expand="md" color="dark" className="navigation">
                <NavBrandLink />
                <ToggleContainer>
                    {(toggled, toggle, toggleOff, toggleOn): React.ReactElement<{}> => 
                        <React.Fragment>
                            <NavbarToggler
                                type="button"
                                onClick={toggle}
                            />
                            <Collapse
                                isOpen={toggled}
                                navbar
                            >
                                <MainMenu onLinkClick={toggleOff} />
                                <TextToSpeech />
                                <AuthMenu />
                            </Collapse>
                        </React.Fragment>}
                </ToggleContainer>
            </Navbar>
        )
    }
}