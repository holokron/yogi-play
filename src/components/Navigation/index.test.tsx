import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Navigation from ".";
import configureStore from "../../store";
import { Provider } from "react-redux";

const store = configureStore();

describe("@components/Navigation", () => {
  it("renders correctly", () => {
    const tree = render(
      <Provider store={store}>
        <MemoryRouter>
          <Navigation />
        </MemoryRouter>
      </Provider>
    );

    expect(tree).toMatchSnapshot();
  });
});
