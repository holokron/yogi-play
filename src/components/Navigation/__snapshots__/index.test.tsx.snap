// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/Navigation renders correctly 1`] = `
<nav
  className="navigation navbar navbar-expand-md navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    <img
      alt="Yogi PLAY"
      className="d-inline-block align-top"
      src="/icons/icon-64x64.png"
      width={30}
    />
     Yogi PLAY
  </a>
  <button
    className="navbar-toggler"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    style={Object {}}
  >
    <ul
      className="mr-auto navbar-nav"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link text-uppercase"
          href="/ulubione"
          onClick={[Function]}
        >
          Ulubione
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link text-uppercase"
          href="https://github.com/holokron/yogi-play/issues/new?labels=sounds&assignee=michalv8"
          rel="noopener noreferrer"
          target="_blank"
        >
          Dodaj dźwięk
        </a>
      </li>
    </ul>
    <ul
      className="mr-auto navbar-nav"
    >
      <form
        className="form-inline"
        noValidate={true}
      >
        <div
          className="input-group"
        >
          <input
            className="input-rounded form-control-sm form-control"
            onChange={[Function]}
            placeholder="szukaj dżwięku"
            type="search"
            value=""
          />
        </div>
      </form>
    </ul>
    <ul
      className="mr-auto navbar-nav"
    >
      <form
        className="form-inline"
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className="input-group"
        >
          <input
            className="input-rounded-left form-control-sm form-control"
            onChange={[Function]}
            placeholder="wpisz tekst"
            type="text"
          />
          <div
            className="input-group-append"
          >
            <button
              aria-label={null}
              className="button-rounded-right btn btn-primary btn-sm"
              onClick={[Function]}
              type="button"
            >
              Czytaj
            </button>
          </div>
        </div>
      </form>
    </ul>
  </div>
</nav>
`;
